### PR TITLE
vmware_guest: answer questions while performing tasks

### PR DESCRIPTION
**Pull request by jeking3**
##### SUMMARY

Some operations that `vmware_guest` is capable of will generate interactive questions.  It is possible to answer those questions automatically if you know what those questions will be.  For example, if you want to change the CD-ROM and the guest has it locked, you can override that by answering a question.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

vmware_guest

##### ADDITIONAL INFORMATION

If `answer` is not provided and an interactive question appears, the default answer provided by VMware is
automatically used as an answer to avoid the task hanging on manual intervention.  The result information contains any questions answered as well as all possible choices.  Playbook authors can use this information to customize the `answer` should the question appear again and the default value is not what is desired.  To test this, I set up an Ubuntu VM running in live boot, which locks the CD.  Here is a sample task attempting cdrom replacement while it is in use, and using the default value (the new output is in the "questions" field of the response:

```
Task:
- hosts: localhost
  tasks:
    - name: Forcibly change the attached virtual optical media, using default answer
      vmware_guest:
        folder: /testvms
        name: testvm_2
        cdrom:
          type: iso
          iso_path: "[datastore1] ISOs/debian-9.8.0-amd64-netinst.iso"

Result:

{  
   "changed":false,
   "failed":true,
   "msg":"Connection control operation failed for disk 'ide0:0'.",
   "op":"reconfig",
   "questions":[  
      {  
         "question":"The guest operating system has locked the CD-ROM door and is probably using the CD-ROM, which can prevent the guest from recognizing media changes. If possible, eject the CD-ROM from inside the guest before disconnecting. Disconnect anyway and override the lock? ",
         "answer":"button.no",
         "choices":[  
            "button.yes",
            "button.no"
         ]
      }
   ]
}
```

If however an answer is provided, the cdrom is forcibly changed:
```
- hosts: localhost
  tasks:
    - name: Forcibly change the attached virtual optical media, using custom answer
      vmware_guest:
        folder: /testvms
        name: testvm_2
        cdrom:
          type: iso
          iso_path: "[datastore1] ISOs/debian-9.8.0-amd64-netinst.iso"
        answer:
          - question: "msg.cdromdisconnect.locked"
            response: "button.yes"
```

Other error cases that are handled:
- multiple answers for the same question is an error
- no answers match the question
- no responses match available choices

In those cases additional details are in the response so the playbook can be corrected.

These changes were manually tested due to the unlikely support in the simulator for handling interactive questions.

### CODE REVIEW

File: lib/ansible/module_utils/vmware.py [hunk 1]
```diff
@@ -1173,3 +1175,94 @@ def find_datastore_cluster_by_name(self, datastore_cluster_name):
...
+            for prop in properties:
+                try:
+                    if '.' in prop:
+                        key, remainder = prop.split('.', 1)
+                        tmp = {}
+                        tmp[key] = self._extract(self._jsonify(getattr(mobj, key)), remainder)
+                        _update(result, tmp)
+                    else:
+                        result[prop] = self._jsonify(getattr(mobj, prop))
+                except AttributeError:
```
Comment (Akasurde) [inline lib/ansible/module_utils/vmware.py]:
```suggestion
                except (AttributeError, KeyError) as e:
```

**Comment by jeking3** [comment]
This should be merged after #47446.  (As you can see, it is included).

File: lib/ansible/module_utils/vmware.py [hunk 2]
```diff
@@ -7,6 +7,8 @@
 __metaclass__ = type
 
 import atexit
+import ansible.module_utils.common._collections_compat as collections_compat
```
Comment (pdellaert) [inline lib/ansible/module_utils/vmware.py]: This should come at the location where other ansible modules are loaded

**Review by pdellaert — CHANGES_REQUESTED** [review]
i like the capability, just want to make sure the imports are ordered correctly

File: lib/ansible/module_utils/vmware.py (hunk 1 above)
Comment (jeking3) [reply lib/ansible/module_utils/vmware.py]: Please ignore that, it is part of #47446 which I said needs to be merged first, then only the changes for this feature will be present.  This change depends on the vmware output schema that #47446 provides so it was branched off #47446.

File: lib/ansible/module_utils/vmware.py (hunk 2 above)
Comment (jeking3) [reply lib/ansible/module_utils/vmware.py]: Okay, I'll make this change once #47446 is merged.

**Pull request closed without merging**